### DISCUSSION TITLE
[Feature Request]: Add Compass/IMU hardware support for more accurate display

### DISCUSSION THREAD

**Discussion by irpyc** [meshtastic/firmware#4357] [Ideas]
**Platform**
NRF52, ESP32

**Description**
It could be really nice to be able to add an hardware compass (IMU), to have a more accurate on device display Compass/Circle.
For example a [Seed Grove - IMU 9DOF ](https://www.seeedstudio.com/Grove-IMU-9DOF-ICM20600-AK09918.html). It don't have many axis, but if only needed for the compass, it is relatively cheap.
Right now, it is easier to glue a mechanical compass to the device and choose to always point Nord, if you want to find the other device without using a Smartphone.

An IMU could also be useful for having a standby mode more reactive than the GPS to changes, and perhaps send notification if the device is moved.

**Comment by s-aluma-r** [comment]
Isn't [this](https://github.com/meshtastic/firmware/commit/0852a170a3b6a10b69fc88963ef267faeda11a07) commit, mostly, what you asked for? Just for the BMX160/RAK12034 and "only" for displaying the actual measured north on the display, not the "wake up detection" you described? This could probably be extended to use other IMUs too.
